zerver/models.py

```diff
@@ -861,6 +861,32 @@ def query_function(type_ids):
     return generic_bulk_cached_fetch(cache_key_function, query_function, type_ids,
                                      id_fetcher=lambda recipient: recipient.type_id)
 
+
+def sew_messages_and_reactions(messages, reactions):
+    # type: (List[Dict[str, Any]], List[Dict[str, Any]]) -> List[Dict[str, Any]]
+    """Given a iterable of messages and reactions stitch reactions
+    into messages.
+    """
+    # Add all messages with empty reaction item
+    for message in messages:
+        message['reactions'] = []
+
+    # Convert list of messages into dictionary to make reaction stitching easy
+    converted_messages = {message['id']: message for message in messages}
+
+    for reaction in reactions:
+        converted_messages[reaction['message_id']]['reactions'].append(
+            reaction)
+
+    values = converted_messages.values()
+
+    # Python 2 returns dict.values as list where as Python 3 returns
+    # as `dict_values`.
+    if isinstance(values, list):
+        return values
+    return list(values)
```

Comment:
The static analysis is complaining about this code; to solve this you should just do for these last like 8 lines

`return list(converted_messages.values())`

Basically, we avoid having any code that behaves differently between Python 2 and Python 3 in a systematic way using this technique.